### PR TITLE
DATAACCESS : Created UserProductLikeRepository, ProductRepository, ProductDiscountRepository, and ProductDetailSizeRepository

### LINKED ISSUE #39: DATAACCESS : Create ProductDetailSizeRepository

### LINKED ISSUE #40: DATAACCESS : Create ProductDiscountRepository

### LINKED ISSUE #41: DATAACCESS : Create ProductRepository

### LINKED ISSUE #42: DATAACCESS : Create UserProductLikeRepository

### PR DESCRIPTION
**Pull request by abdulazizorazbaev**
Closes #39 
Closes #40 
Closes #41 
Closes #42

### CODE REVIEW

File: src/Eclo.DataAccess/Repositories/Products/ProductDetailSizeRepository.cs [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+public class ProductDetailSizeRepository : BaseRepository, IProductDetailSizeRepository
+{
+    public async Task<long> CountAsync()
+    {
+        try
+        {
+            await _connection.OpenAsync();
+            string query = "SELECT COUNT(*) FROM product_detail_sizes";
+            var result = await _connection.QuerySingleAsync<long>(query);
+            return result;
```
Comment (ShohruhUzDev) [inline src/Eclo.DataAccess/Repositories/Products/ProductDetailSizeRepository.cs:17]: Return boshqa kodlardan ajralib tursin.

File: src/Eclo.DataAccess/Repositories/Products/ProductDetailSizeRepository.cs [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+            await _connection.CloseAsync();
+        }
+    }
+
+    public async Task<int> CreateAsync(ProductDetailSize entity)
+    {
+        try
+        {
+            await _connection.OpenAsync();
+            string query = "INSERT INTO public.product_detail_sizes (product_detail_id, size, quantity, created_at, updated_at) " +
```
Comment (ShohruhUzDev) [inline src/Eclo.DataAccess/Repositories/Products/ProductDetailSizeRepository.cs:34]: Bush joy quyilishi kerak.

File: src/Eclo.DataAccess/Repositories/Products/ProductDetailSizeRepository.cs [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+                "VALUES (@ProductDetailId, @Size, @Quantity, @CreatedAt, @UpdatedAt);";
+            var result = await _connection.ExecuteAsync(query, entity);
```
Comment (ShohruhUzDev) [inline src/Eclo.DataAccess/Repositories/Products/ProductDetailSizeRepository.cs:36]: var result variable tepa va pastki kodlardan ajralib tursin.

File: src/Eclo.DataAccess/Repositories/BaseRepository.cs [hunk 4]
```diff
@@ -8,6 +8,6 @@ public class BaseRepository
     public BaseRepository()
     {
         Dapper.DefaultTypeMap.MatchNamesWithUnderscores = true;
-        this._connection = new NpgsqlConnection("Host=localhost; Port=5432; Database=eclo-db; User Id=postgres; Password=1;");
+        this._connection = new NpgsqlConnection("Host=localhost; Port=5432; Database=eclo-db; User Id=postgres; Password=12345;");
```
Comment (ShohruhUzDev) [inline src/Eclo.DataAccess/Repositories/BaseRepository.cs:11]: bitta pastki qator tashlanib tab quyilsin.

File: src/Eclo.DataAccess/Repositories/Products/ProductDetailSizeRepository.cs (hunk 3 above)
Comment (abdulazizorazbaev) [reply src/Eclo.DataAccess/Repositories/Products/ProductDetailSizeRepository.cs:39]: All lines like this have been changed or rewritten

File: src/Eclo.DataAccess/Repositories/BaseRepository.cs (hunk 4 above)
Comment (abdulazizorazbaev) [reply src/Eclo.DataAccess/Repositories/BaseRepository.cs]: Resolved

File: src/Eclo.DataAccess/Repositories/Products/ProductDetailSizeRepository.cs (hunk 2 above)
Comment (abdulazizorazbaev) [reply src/Eclo.DataAccess/Repositories/Products/ProductDetailSizeRepository.cs]: Done

File: src/Eclo.DataAccess/Repositories/Products/ProductDetailSizeRepository.cs (hunk 1 above)
Comment (abdulazizorazbaev) [reply src/Eclo.DataAccess/Repositories/Products/ProductDetailSizeRepository.cs]: This type of problems have been done

File: src/Eclo.DataAccess/Repositories/Products/ProductRepository.cs [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+    public async Task<(long ItemsCount, IList<ProductViewModel>)> SearchAsync(string search, PaginationParams @params)
+    {
+        try
+        {
+            await _connection.OpenAsync();
+
+            string query = $"SELECT * FROM product_view WHERE product_name ILIKE @search ORDER BY id DESC " +
+                $"OFFSET {@params.GetSkipCount()} LIMIT {@params.PageSize}";
+
+            var result = (await _connection.QueryAsync<ProductViewModel>(query, new { search = "%" + search + "%" })).ToList();
```
Comment (ShohruhUzDev) [inline src/Eclo.DataAccess/Repositories/Products/ProductRepository.cs:127]: 121 tadan oshgan

File: src/Eclo.DataAccess/Repositories/Products/ProductRepository.cs [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+
+    public async Task<int> UpdateAsync(long id, Product entity)
+    {
+        try
+        {
+            await _connection.OpenAsync();
+
+            string query = $"UPDATE public.products " +
+                $"SET brand_id=@BrandId, sub_category_id=@SubCategoryId, name=@Name, unit_price=@UnitPrice, " +
+                $"description=@Description, created_at=@CreatedAt, updated_at=@UpdatedAt " +
```
Comment (ShohruhUzDev) [inline src/Eclo.DataAccess/Repositories/Products/ProductRepository.cs:149]: tab

File: src/Eclo.DataAccess/Repositories/Products/UserProductLikeRepository.cs [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+
+    public async Task<int> CreateAsync(UserProductLike entity)
+    {
+        try
+        {
+            await _connection.OpenAsync();
+            
+            string query = "INSERT INTO public.user_product_likes " +
+                "(user_id, product_id, is_liked, created_at, updated_at) " +
+                "VALUES (@UserId, @ProductId, @IsLiked, @CreatedAt, @UpdatedAt);";
```
Comment (ShohruhUzDev) [inline src/Eclo.DataAccess/Repositories/Products/UserProductLikeRepository.cs:38]: tab

File: src/Eclo.DataAccess/Repositories/Products/UserProductLikeRepository.cs [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+
+    public async Task<int> UpdateAsync(long id, UserProductLike entity)
+    {
+        try
+        {
+            await _connection.OpenAsync();
+            
+            string query = $"UPDATE public.user_product_likes " +
+                $"SET user_id=@UserId, product_id=@ProductId, is_liked=@IsLiked, created_at=@CreatedAt, updated_at=@UpdatedAt" +
+                $"WHERE id = {id};";
```
Comment (ShohruhUzDev) [inline src/Eclo.DataAccess/Repositories/Products/UserProductLikeRepository.cs:102]: tab

**Review by Maxsetovich — APPROVED** [review]
Completed products repository

**Pull request merged**